rice-proto/src/candidate.rs

```diff
@@ -560,7 +560,7 @@ mod parse {
                 take_while1::<_, _, nom::error::Error<_>>(is_part_of_byte_string)(s)
                     .map_err(|_| ParseCandidateError::BadExtension)?;
 
-            if let Some(expected_next) = expected_next {
+            if let Some(expected_next) = expected_next.take() {
                 if ext_key != expected_next {
                     return Err(ParseCandidateError::BadExtension);
                 }
```

Comment:
This is dead code now, I commented it out here and the tests don't complain. Was there a special reason for this test?

Reply:
e.g. the raddr, and rport attributes.

Reply:
They are not stored in the extension vector, no. But they are retrieved at the same time as the extension vector.